homeassistant/components/netatmo/media_source.py

```diff
@@ -0,0 +1,140 @@
+"""Netatmo Media Source Implementation."""
+import datetime as dt
+import mimetypes
+from typing import Optional, Tuple
+
+from homeassistant.components.media_player.errors import BrowseError
+from homeassistant.components.media_source.const import MEDIA_MIME_TYPES
+from homeassistant.components.media_source.error import Unresolvable
+from homeassistant.components.media_source.models import (
+    BrowseMedia,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DATA_CAMERAS, DATA_EVENTS, DOMAIN, MANUFACTURER
+
+
+async def async_get_media_source(hass: HomeAssistant):
+    """Set up Netatmo media source."""
+    return NetatmoSource(hass)
+
+
+class NetatmoSource(MediaSource):
+    """Provide Netatmo camera recordings as media sources."""
+
+    def __init__(self, hass: HomeAssistant):
+        """Initialize Netatmo source."""
+        super().__init__(DOMAIN)
+        self.hass = hass
+        self.events = self.hass.data[DOMAIN][DATA_EVENTS]
+
+    @callback
+    def async_parse_identifier(self, item: MediaSourceItem) -> Tuple[str, str]:
+        """Parse identifier."""
+        if not item.identifier:
+            source_dir_id = "events"
+            location = ""
+        else:
+            source_dir_id, location = item.identifier[
+                item.identifier.startswith("/") and 1 :
```

Comment:
Wuuut. Please extract this in a variable.

Reply:
I've switched to `lstrip("/")`. More readable now.